biz.aQute.repository/src/aQute/bnd/deployer/repository/LocalIndexedRepo.java

```diff
@@ -69,6 +105,30 @@ public synchronized void setProperties(Map<String,String> map) {
 			throw new IllegalArgumentException(String.format(
 					"Cannot create repository cache: '%s' already exists but is not directory.",
 					cacheDir.getAbsolutePath()));
+
+		setIncludePolicy(map);
+	}
+
+	private void setIncludePolicy(Map<String,String> props) {
+		String policy = props.get(PROP_FILE_INCLUDE_POLICY);
+		if (policy != null && !policy.isEmpty()) {
+			try {
+				FileIncludePolicy spec = FileIncludePolicy.valueOf(policy);
+				includePolicy = spec.pattern;
+			}
+			// this means the policy was not recognized, so try to interpret it
+			// as a regex
+			catch (IllegalArgumentException e) {
+				includePolicy = Pattern.compile(policy);
```

Comment:
You must handle the `PatternSyntaxException` here. You can't just let it bubble up. An error must be reported as the statement is malformed.